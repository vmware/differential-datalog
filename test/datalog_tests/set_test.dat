dump set_test::SetTransforms;
dump set_test::SetFolds;
