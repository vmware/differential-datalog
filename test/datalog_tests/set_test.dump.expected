set_test::SetTransforms{.description = "all(contains(3))", .s = []}
set_test::SetTransforms{.description = "any(contains(3))", .s = [(0, [0, 1, 2]), (1, [2, 3, 4]), (2, [4, 5, 6]), (3, [6, 7, 8]), (4, [8, 9, 10])]}
set_test::SetTransforms{.description = "arg_max(max)", .s = [(4, [8, 9, 10])]}
set_test::SetTransforms{.description = "arg_min(min)", .s = [(0, [0, 1, 2])]}
set_test::SetTransforms{.description = "filter(contains 2)", .s = [(0, [0, 1, 2]), (1, [2, 3, 4])]}
set_test::SetTransforms{.description = "filter_map(==1, :=100)", .s = [(100, [2, 3, 4])]}
set_test::SetTransforms{.description = "find(contains(3))", .s = [(1, [2, 3, 4])]}
set_test::SetTransforms{.description = "map(push 100)", .s = [(0, [0, 1, 2, 100]), (1, [2, 3, 4, 100]), (2, [4, 5, 6, 100]), (3, [6, 7, 8, 100]), (4, [8, 9, 10, 100])]}
set_test::SetFolds{.description = "fold(+)", .a = 10}
set_test::SetFolds{.description = "fold(fold(+))", .a = 75}
